packages/typescript/types/index.d.ts

```diff
@@ -41,7 +41,7 @@ export interface RollupTypescriptPluginOptions {
 type ElementType<T extends Array<any>> = T extends (infer U)[] ? U : never;
 
 export type TransformerStage = keyof CustomTransformers;
-type StagedTransformerFactory<T extends TransformerStage> = ElementType<CustomTransformers[T]>;
+type StagedTransformerFactory<T extends TransformerStage> = ElementType<CustomTransformers[]>;
```

Comment:
This seems to be an error. This change means that it isn't the list of a single transform step anymore, but an array of an array of transforms.

The type of `CustomTransformers` is:

```typescript
export interface CustomTransformers {
    /** Custom transformers to evaluate before built-in .js transformations. */
    before?: (TransformerFactory<SourceFile> | CustomTransformerFactory)[];
    /** Custom transformers to evaluate after built-in .js transformations. */
    after?: (TransformerFactory<SourceFile> | CustomTransformerFactory)[];
    /** Custom transformers to evaluate after built-in .d.ts transformations. */
    afterDeclarations?: (TransformerFactory<Bundle | SourceFile> | CustomTransformerFactory)[];
}
```

So `CustomTransformers[T]` already is an array. Making another array from it, removes the error but is probably semantically incorrect.

See https://github.com/rollup/plugins/pull/633#issuecomment-721250077 for a discussion of alternative solutions.

Reply:
I wasn’t quite sure what was happening here so I just tried to make it work. I’ve used the first solution you suggested to fix the issue.